wayland.c

```diff
@@ -75,11 +79,22 @@ static void create_output(struct mako_state *state,
 	output->global_name = global_name;
 	output->wl_output = wl_output;
 	output->scale = 1;
+
+	if (wl_list_empty(&state->outputs)) {
+		recreate_surface = true;
+	}
+
 	wl_list_insert(&state->outputs, &output->link);
 
 	wl_output_set_user_data(wl_output, output);
 	wl_output_add_listener(wl_output, &output_listener, output);
 	get_xdg_output(output);
+	if (recreate_surface) {
+		// We had no outputs, force our surfaces to redraw
+		wl_list_for_each(surface, &output->state->surfaces, link) {
+			send_frame(surface);
```

Comment:
I think this should use `set_dirty` instead of sending the frame directly, just to keep the code path consistent. It should have the same effect of immediately drawing a frame in this case since the underlying surfaces will have been gone.

(Are we cleaning them up properly when the last output goes away?)

Reply:
Done.

When the last output goes away, one of the buffers may still be marked as busy, however the compositor should release it soon after. When restarting with one or more outputs, the get_next_buffer() code should clean it up.